### ISSUE TITLE
annotation icons are hidding content

### ISSUE THREAD

**Issue by ousia** [sumatrapdfreader/sumatrapdf#1600]
@ousia 
re: your comment "annotation icons are huge"
can you supply an example since SumatraPDF is tied by the way MuPDF shows those icons
Thus they are a fixed size which will appear larger than normal if the page is smaller than normal
![image](https://user-images.githubusercontent.com/12754913/83195139-a8760e00-a131-11ea-97fe-8e5810dc957e.png)

_Originally posted by @GitHubRulesOK in https://github.com/sumatrapdfreader/sumatrapdf/issues/1557#issuecomment-635616213_

**Comment by ousia** [comment]
Sorry, @GitHubRulesOK, the comment above didn’t work as expected.

Latest pre-release (`SumatraPDF-prerel-12665.exe`) displays exactly the same as MuPDF:

![gl-attach](https://user-images.githubusercontent.com/203767/83334572-c82f4280-a2a7-11ea-80c7-c8d983e4ea58.png)

Acrobat doesn’t hide characters:

![acroread-attach](https://user-images.githubusercontent.com/203767/83334574-c9606f80-a2a7-11ea-9104-f2194dea6912.png)

Nor does Evince:

![evince-attach](https://user-images.githubusercontent.com/203767/83334575-c9f90600-a2a7-11ea-801c-50190ee59d4a.png)

**Comment by ousia** [comment]
But this also happens with other annotations icons.

Compare SumatraPDF display:

![just-sumatra](https://user-images.githubusercontent.com/203767/83334835-738cc700-a2a9-11ea-8fa9-15eec9d7f4c0.png)

To what Acrobat displays:

![just-acrobat](https://user-images.githubusercontent.com/203767/83334837-74255d80-a2a9-11ea-8811-5a9c62ecce6e.png)

This is hiding characters.

**Comment by ousia** [comment]
I know that the issue is upstream.

Now the real issue is that SumatraPDF is problematic (because MuPDF is) displaying annotation icons not generated with MuPDF itself.

And worst of all, annotation icons are printed (but this requires a new issue).

**Comment by GitHubRulesOK** [comment]
@kjk 
Looks like MuPDF is using icons at least twice the size compared to previous
I would suggest the rect be halved or fixed at about 10 pts and for western users best scaled towards upper right
![image](https://user-images.githubusercontent.com/12754913/83587006-db5f3e00-a545-11ea-88ff-af0d81377ada.png)
There could be a related issue with oversize stamps 
but that is best treated as a seperate scalar value

**Comment by kjk** [comment]
Reported to mupdf: https://bugs.ghostscript.com/show_bug.cgi?id=702544

**Comment by GitHubRulesOK** [comment]
@ousia 
Whilst it does not answer the original problem, that Icons take up more space (partly due to no transparency, partly due to scale).

It is now possible with the latest prerelease to drag the icon to clear it away from existing objects.
There is a downside, that it will alter the embedded icon to the new ones, so a file will now appear identical in version 3.1.2

![image](https://user-images.githubusercontent.com/12754913/87233709-c1750e80-c3c1-11ea-9001-6e1c63e7e916.png)

**Comment by GitHubRulesOK** [comment]
Just an additional related observation on the differences in icon behaviour.
SumatraPDF (i.e. MuPDF engine) does not mess with the position or relative scale of the 18x18 pt Icon so it appears bigger as you zoom in. 
However Acrobat will size the Icon per screen pixels thus the icon appears relatively smaller as you zoom in.
Both methods can have merit in different circumstances, but most users may favour the Adobe approach?
![image](https://user-images.githubusercontent.com/12754913/87873373-12d05f80-c9b9-11ea-8b68-470b5ac54c60.png)

**Comment by ousia** [comment]
> Just an additional related observation on the differences in icon behaviour.
> SumatraPDF (i.e. MuPDF engine) does not mess with the position or relative scale of the 18x18 pt Icon so it appears bigger as you zoom in.
> However Acrobat will size the Icon per screen pixels thus the icon appears relatively smaller as you zoom in.
> Both methods can have merit in different circumstances, but most users may favour the Adobe approach?
> ![image](https://user-images.githubusercontent.com/12754913/87873373-12d05f80-c9b9-11ea-8b68-470b5ac54c60.png)

@GitHubRulesOK,

many thanks for your reply.

I think _Adobe_ may be inconsistent here.

From the [“Annotation flags” table](https://www.adobe.com/content/dam/acom/en/devnet/pdf/PDF32000_2008.pdf#nameddest=G11.2096191) from the PDF spec (`NoZoom` value):

> (PDF 1.3) If set, do not scale the annotation’s appearance to match the magnification of the page. The location of the annotation on the page (defined by the upper-left corner of its annotation rectangle) shall remain fixed, regardless of the page magnification.

Explanation [on the following page](https://www.adobe.com/content/dam/acom/en/devnet/pdf/PDF32000_2008.pdf#page=394):

> If the NoZoom flag is set, the annotation shall always maintain the same fixed size on the screen and shall be unaffected by the magnification level at which the page itself is displayed. Similarly, if the NoRotate flag is set, the annotation shall retain its original orientation on the screen when the page is rotated (by changing the Rotate entry in the page object; see [7.7.3, “Page Tree”](https://www.adobe.com/content/dam/acom/en/devnet/pdf/PDF32000_2008.pdf#nameddest=M6.9.35899.2Heading.392.Page.Tree)).

**Comment by GitHubRulesOK** [comment]
@ousia 
Thanks for the heads up it looks like MuPDF respects the no scale setting BUT does not maintain rotation well, The active zone is kept but the icon is spun !!
![image](https://user-images.githubusercontent.com/12754913/87874782-93488d80-c9c4-11ea-8fcd-400160f67b25.png)

**Comment by ousia** [comment]
> @ousia
> Thanks for the heads up it looks like MuPDF respects the NoZoom flag ?

@GitHubRulesOK,

to translate binary positions to values (sorry if this is obvious to you, it isn’t for me).

00000000 = 0
00000001 = 1
00000010 = 2
00000011 = 3
00000100 = 4

 So `/F 4` is set to `Print` (according to the [“Annotation flags” table](https://www.adobe.com/content/dam/acom/en/devnet/pdf/PDF32000_2008.pdf#nameddest=G11.2096191) [and in this case the annotation should be printed because it lacks an appearance dictionary]).

`/F 4` refers to value, but the bit position (see above) is the third one. So `Print` is set for the annotation. 

What it seems to happen here is that _Acrobat_ doesn’t seem to respect the `NoZoom` flag. Annotations should be zoomed, since no `NoZoom` flag is really set here.

> However SumatraPDF does not maintain rotation well.

Unless you modified the PDF document, `NoRotate` flag isn’t set for the annotation (that would read `/F 16` [or higher]).

To summarize: from the provided PDF document, one cannot tell whether MuPDF follows the spec regarding annotation flags. _Acrobat_ might be misbehaving here.

BTW, I guess these things should be implemented upstream (in MuPDF).

**Comment by GitHubRulesOK** [comment]
@ousia 
digging deeper into those flags the problem here does appear to lie with MuPDF not respecting /F 8 
(a related issue re SumatraPDF and rotation is now opened as #1702 )

**Comment by GitHubRulesOK** [comment]
At present with respect to appearance here are acrobat and SumatraPDF at 100% 25% and 400% zoom all now seem to agree in scaling so am closing without any example to prove there is still a related issue about size of icons.

![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/17efe840-348c-4bfd-be3f-ceda36ef5323)
25% zoom
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/86136c34-af20-4830-ad7a-0eef42ce0760)
400% zoom
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/52ef3379-440d-4c6b-9c5c-301313d7f6be)
compared at 100% with Edge
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/754e94c7-8ffc-440a-b497-fabd1a1d5d43)

**Comment by GitHubRulesOK** [comment]
sample of above scaling problem
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/adbb9913-18ff-4f67-ac4c-0bfcec46f0fd)

**Comment by kr4z33** [comment]
Having trouble parsing the crowded annotations in a file I am working with - the mupdf report referenced https://github.com/sumatrapdfreader/sumatrapdf/issues/1600#issuecomment-654069988 is "resolved fixed". But the screenshot sample below shows the annotations could be presented in a clearer way - dragging them around is not fully satisfying workaround, because the accurate position of the notes is also of relevance in my particular case. It would be nice to have some control on the icon size to reduce or eliminate the overlapping to allow for parsing through the notes in cases like this. 

<img width="577" height="328" alt="Image" src="https://github.com/user-attachments/assets/b9872211-214a-4657-976a-efb49735babd" />

**Comment by GitHubRulesOK** [comment]
@kr4z33 
I was not Adobe staff last century but the idea of notes in a markup (Which I did Support for many products) was intended for copywriters or drawing REDLINE e.g. a few comments about errors,
They were not intended as every dot on a circuit board with a metadata history (computers at that time simply would not have coped with the density you show)


What has "happened" is they should be scalar so as you zoom in ideally there would be more space between them but that's not the way they always work.
> Acrobat will size the Icon per screen pixels  or different

<img width="751" height="288" alt="Image" src="https://github.com/user-attachments/assets/36779823-0435-4001-b22e-22a726e14b4f" />
 

In MuPDF
when you zoom in they retain a fixed point size

<img width="674" height="477" alt="Image" src="https://github.com/user-attachments/assets/f6b7ef68-7fc4-4ac9-9beb-6cf629bad4cf" />

Other apps may be more as expected

[AntiAliasExampleannot2.pdf](https://github.com/user-attachments/files/26385640/AntiAliasExampleannot2.pdf)

<img width="1015" height="501" alt="Image" src="https://github.com/user-attachments/assets/3b60c30b-912f-4f60-a3eb-7006d6d6c034" />

**Comment by kr4z33** [comment]
Thanks for taking a look at this - I do agree this is likely well beyond the original intents of the text annotation feature, but here we are to deal with the overloading. 

The context in my particular example are that these are annotations on a pdf formatted satellite image documenting features on the ground, as recorded on a mobile device in the field.

This kind of issue also tends to come up in annotations on construction plans, typically exchanged in PDF format with large paper size and mostly graphical content.

**Comment by GitHubRulesOK** [comment]
@kjk the TODAY release of 1.28 with the long ago Mupdf fix has changed prior behaviour such that in "theory" the OP issue was addressed 

<img width="610" height="242" alt="Image" src="https://github.com/user-attachments/assets/a6990b5c-5bb2-4c0a-b30c-dd263e187b3c" />


SumatraPDF inherited that fix and thus may best consider the primary issue was fixed, and any new need samples to be addressed differently.
<img width="507" height="207" alt="Image" src="https://github.com/user-attachments/assets/ab2de9e6-df63-4928-b12e-898c82c2bb78" />

**Comment by kjk** [comment]
Closing because the primary annotation icon sizing/rendering issue appears fixed in current MuPDF/SumatraPDF builds.

There may still be separate usability issues for PDFs with very dense note annotations, but those should be tracked with concrete current samples. If this is still a problem, please open a new issue with:

```
1. The exact SumatraPDF version
2. A small PDF that demonstrates the overlap/problem
3. A screenshot showing the expected vs actual annotation display
```
